### PR TITLE
standardize connection variable names

### PR DESCRIPTION
**Pull request by jamescassell**
##### SUMMARY

This is PR1 per discussion in the core meeting today.  Enables use of `_user` and `_password` version of vars for all connection types where `_username` or `_pass` versions were present.

Partial fix for #51757

##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request

##### COMPONENT NAME

connection_plugins
docs

### CODE REVIEW

File: docs/docsite/rst/user_guide/become.rst [hunk 1]
```diff
@@ -73,7 +73,7 @@ ansible_become_method
 ansible_become_user
     set the user you become through privilege escalation; does not imply ``ansible_become: yes``
 
-ansible_become_pass
+ansible_become_password
```
Comment (jborean93) [inline docs/docsite/rst/user_guide/become.rst:76]: I don't think this is correct, there is work to introduce become plugins where you can set variables in the config like connection options but right now this is magic done in the play context.

Comment (jamescassell) [reply docs/docsite/rst/user_guide/become.rst:76]: At the core meeting today, it was agreed to change all `ansible_*_pass` vars to `ansible_*_password` and all `ansible_*_username` vars to `ansible_*_user`.  This particular change is just a docs change.  You already can use `ansible_become_password` or `ansible_become_pass`, though I'm unsure of which will win today if you specify both.

Comment (jborean93) [reply docs/docsite/rst/user_guide/become.rst:76]: Ah ok, I see https://github.com/ansible/ansible/blob/devel/lib/ansible/constants.py#L166 is where the current magic happens. I wasn't aware that `ansible_become_password` worked today but good to know.

**Review by samccann — APPROVED** [review]
Docs LGTM

**Review by Qalthos — APPROVED** [review]
Networking-wise, docs & plugins look good

**Comment by gundalow** [comment]
@jamescassell Thank you!

**Comment by samccann** [comment]
Thanks @jamescassell for the Ansible doc change!

**Comment by bcoca** [comment]
@jamescassell @samccann @gundalow  @Qalthos  we might need to revert these changes, passwords are not handled by the config system on all the plugins updated, so the new variables will not work.

**Comment by jamescassell** [comment]
Does that force us to change the vars wholesale? Either Breaking existing setups, or keeping the status quo?

**Comment by bcoca** [comment]
the change you made cannot work ... not sure how reverting that is going to break existing setups?

**Comment by jamescassell** [comment]
@bcoca can you point to one of the the plugins that doesn't work... maybe I can fix it?

**Comment by bcoca** [comment]
@jamescassell you changed the docs but didnt change the connection plugin options to actually use the doc config

**Comment by jamescassell** [comment]
Ok, thanks for the explanation. I had been under the impression that the plugin options were generated from the docs. I'll see about fixing it.

**Comment by bcoca** [comment]
most are, but many ssh/paramiko were not all ported due to them being overloaded on other plugins and they also are present in command line

**Comment by jamescassell** [comment]
The only place that `ansible_paramiko_pass` is defined is in `lib/ansible/plugins/connection/paramiko_ssh.py`

I'm guessing that the only one causing issues is `ansible_ssh_password` because `ansible_ssh_pass` is hardcoded in several places.  AFAICS, the other ones are only defined in the respective plugin docs fragment.  `ansible_ssh_pass` is also in `lib/ansible/constants.py`, `lib/ansible/playbook/play_context.py`, `lib/ansible/plugins/action/synchronize.py`

Seems that we should be able to update those 3 uses to also support `ansible_ssh_password`, or otherwise remove the `ansible_ssh_pass` from those uses and/or point folks towards `ansible_password` instead?

**Comment by bcoca** [comment]
depends, not all plugins are using the same system, you have to go per plugin check out which ones use play_context vs get_option. Also syncronize is probably wrong unless you are specifically using ssh connection plugin

**Pull request merged**